### PR TITLE
setPedro/dapps

### PR DESCRIPTION
**Pull request by setPedro**

### CODE REVIEW

File: src/routes/balances.rs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    },
+    client::SolanaMirrorClient,
+    utils::get_rpc,
+    Error,
+};
+
+#[get("/balances/<address>?<positions>")]
+pub async fn accounts_handler(
+    address: &str,
+    positions: Option<bool>,
```
Comment (config-json) [inline src/routes/balances.rs:20]: This should be a more generic param, we're going to have dapps that are not necessarily LPs

File: src/routes/balances.rs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+                Error::TooManyRequests => Status::TooManyRequests,
+                _ => Status::InternalServerError,
+            };
+            return Err(status_code);
+        }
+    };
+
+    let (position_accounts, filtered_parsed_accounts): (Vec<_>, Vec<_>) = parsed_accounts
+        .into_iter()
+        .partition(|account| account.balance.amount == "1");
```
Comment (config-json) [inline src/routes/balances.rs:44]: Looks like this can be a util

File: src/routes/balances.rs [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+    // return only accounts if positions is explicitely false
```
Comment (config-json) [inline src/routes/balances.rs:46]: [nit] remove this comment

File: lib/src/balances/dapps/raydium/mod.rs [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+        // Both tokens are present
+        amount_a = liquidity_f64 * (1.0 / sqrt_price_current - 1.0 / sqrt_price_upper);
+        amount_b = liquidity_f64 * (sqrt_price_current - sqrt_price_lower);
+    } else {
+        // There is only token A (base token)
+        amount_a = 0.0;
+        amount_b = liquidity_f64 * (sqrt_price_upper - sqrt_price_lower);
+    }
+
+    (amount_a.ceil(), amount_b.ceil())
```
Comment (config-json) [inline lib/src/balances/dapps/raydium/mod.rs:181]: We should round not ceil/truncate

File: lib/src/balances/dapps/raydium/mod.rs [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+    let sqrt_price_upper = (1.0001f64.powi(tick_upper) as f64).sqrt();
+
+    let liquidity_f64 = liquidity as f64;
+
+    let amount_a;
+    let amount_b;
+
+    if sqrt_price_current <= sqrt_price_lower {
+        // There is only token B (quote token)
+        amount_a = liquidity_f64 * (1.0 / sqrt_price_lower - 1.0 / sqrt_price_upper);
```
Comment (config-json) [inline lib/src/balances/dapps/raydium/mod.rs:169]: Extract into util we will need it for Orca for instance

File: lib/src/balances/dapps/raydium/mod.rs [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+        amount_b = 0.0;
+    } else if sqrt_price_current < sqrt_price_upper {
+        // Both tokens are present
+        amount_a = liquidity_f64 * (1.0 / sqrt_price_current - 1.0 / sqrt_price_upper);
+        amount_b = liquidity_f64 * (sqrt_price_current - sqrt_price_lower);
```
Comment (config-json) [inline lib/src/balances/dapps/raydium/mod.rs:174]: Extract into util

File: lib/src/balances/dapps/raydium/mod.rs [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+fn calculate_token_amounts(
+    liquidity: u128,
+    tick_lower: i32,
+    tick_upper: i32,
+    sqrt_price_x64: u128,
+) -> (f64, f64) {
+    let sqrt_price_current = (sqrt_price_x64 as f64) / (1u128 << 64) as f64;
+
+    let sqrt_price_lower = (1.0001f64.powi(tick_lower) as f64).sqrt();
+    let sqrt_price_upper = (1.0001f64.powi(tick_upper) as f64).sqrt();
```
Comment (config-json) [inline lib/src/balances/dapps/raydium/mod.rs:160]: Extract this calculation into a util eg `getSqrtPriceFromTick`

**Review by config-json — COMMENTED** [review]
LGTM, left some comments. We need to conserve the `/accounts` endpoint for now until we change it in the rest of the codebase

File: src/routes/balances.rs (hunk 2 above)
Comment (setPedro) [reply src/routes/balances.rs:44]: overkill

**Review by config-json — DISMISSED** [review]

**Review by config-json — APPROVED** [review]

**Pull request merged**